yt_dlp/downloader/youtube_live_chat.py

```diff
@@ -31,13 +32,116 @@ def real_download(self, filename, info_dict):
+                    offset = int(replay_chat_item_action['videoOffsetTimeMsec'])
+                processed_fragment.extend(
+                    json.dumps(action, ensure_ascii=False).encode('utf-8') + b'\n')
+            if offset is not None:
+                continuation_id = try_get(
+                    live_chat_continuation,
+                    lambda x: x['continuations'][0]['liveChatReplayContinuationData']['continuation'])
+            self._append_fragment(ctx, processed_fragment)
+            return continuation_id, offset
+
+        def parse_live_timestamp(action):
+            action_content = (
+                try_get(action, lambda x: x['addChatItemAction'], dict)
+                or try_get(action, lambda x: x['addLiveChatTickerItemAction'], dict)
+                or try_get(action, lambda x: x['addBannerToLiveChatCommand'], dict)
+            )
+            if action_content is None or type(action_content) != dict:
+                return None
+            item = (
+                try_get(action_content, lambda x: x['item'], dict)
+                or try_get(action_content, lambda x: x['bannerRenderer'], dict)
+            )
+            if item is None:
+                return None
+            renderer = (
+                # text
+                try_get(item, lambda x: x['liveChatTextMessageRenderer'], dict)
+                or try_get(item, lambda x: x['liveChatPaidMessageRenderer'], dict)
+                or try_get(item, lambda x: x['liveChatMembershipItemRenderer'], dict)
+                or try_get(item, lambda x: x['liveChatPaidStickerRenderer'], dict)
+                # ticker
+                or try_get(item, lambda x: x['liveChatTickerPaidMessageItemRenderer'], dict)
+                or try_get(item, lambda x: x['liveChatTickerSponsorItemRenderer'], dict)
+                # banner
+                or try_get(item, lambda x: x['liveChatBannerRenderer'], dict)
+            )
+            if renderer is None:
+                return None
+            parent_item = (
+                try_get(
+                    renderer,
+                    lambda x: x['showItemEndpoint']['showLiveChatItemEndpoint']['renderer'], dict)
+                or try_get(renderer, lambda x: x['contents'], dict)
+            )
+            if parent_item:
+                renderer = (
+                    try_get(parent_item, lambda x: x['liveChatTextMessageRenderer'], dict)
+                    or try_get(parent_item, lambda x: x['liveChatPaidMessageRenderer'], dict)
+                    or try_get(parent_item, lambda x: x['liveChatMembershipItemRenderer'], dict)
+                    or try_get(parent_item, lambda x: x['liveChatPaidStickerRenderer'], dict)
+                )
+            if renderer:
+                timestamp_usec = try_get(renderer, lambda x: x['timestampUsec'], str)
+                if timestamp_usec:
+                    return int(int(timestamp_usec) / 1000)
```

Comment:
```suggestion
                timestamp_usec = int_or_none(try_get(renderer, lambda x: x['timestampUsec'], str), 1000)
```

Reply:
Thanks, didn't know about `int_or_none`. I'll do this a bit differently together with `try_get` refactoring